automator/automator.py

```diff
@@ -162,9 +141,17 @@ def tracking(self, subarray_name):
         We don't take any action immediately, but we do want to set a timer to check
         back later to see if we are ready to process.
         """
-        nshot = self.get_nshot(subarray_name)
-        log.info('{} in tracking state with nshot = {}'.format(subarray_name, nshot))
 
+        # If a sequence of primary time observations has ended (ie, this current
+        # track is not a primary time track), then process right away:
+        if redis_util.primary_sequence_end(self.redis_server, subarray_name):
+            # Disable recording here for the subarray, to avoid a race condition.
+            # Is there a better way of doing this?
+            redis_util.disable_recording(self.redis_server, subarray_name)
+            log.info('Primary sequence has ended, proceeding to processing')
+            self.maybe_start_processing()
```

Comment:
OK so my thinking here is: if we have been recording primary time tracks, the earliest indication that we are done with the primary time sequence of tracks is when we start a non-primary time track. When this happens, we know with certainty that the coordinator is not going to record the track that has just started, and so we can ignore it and go straight to processing, and not worry about the timer. 

To answer your question, no - you're right, checking again would be redundant. So instead (if we keep this code here) I'd do it as follows:

If this is the end of a primary time sequence of recordings, then process immediately. 
Else start a timer and check to see if we should process afterwards. 

We could also just leave this out entirely, and we would process anyway after `DWELL` seconds, maybe this would be simpler/better to do?

Reply:
I'm not entirely sure I understand the expected control flow, but I think if this logic is only supposed to run in super rare cases, I'd leave it out, because you'll never notice if there's a bug in it until it's too late. Or if you do want to take action here, you can just add in a maybe_start_processing call, and make sure the logic inside maybe_start_processing is fine to be called whenever, that's pretty safe. Ie including the call to "disable_recording".

Reply:
OK sounds good, have removed it for now.